### PR TITLE
py-reportseff: add new package

### PR DESCRIPTION
**Pull request by erimar77**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-reportseff/package.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-reportseff/package.py]:
```suggestion
# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
```

**Review by tldahlgren — COMMENTED** [review]
Confirmed the sha256.

**Comment by tldahlgren** [comment]
Adding python gurus as reviewers.

File: var/spack/repos/builtin/packages/py-reportseff/package.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+class PyReportseff(PythonPackage):
+    """A python script for tabular display of slurm efficiency information."""
+
+    homepage = "https://github.com/troycomi/reportseff"
+    pypi = "reportseff/reportseff-2.7.2.tar.gz"
+
+    version("2.7.2", sha256="63cf99ceb0111de511636b214ade937c6c1b8444531e8026dbc34ddf92049c41")
+
+    depends_on("python@3.7:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-reportseff/package.py]:
```suggestion
    depends_on("python@3.7:3", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-reportseff/package.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    depends_on("py-poetry-core", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-reportseff/package.py]:
```suggestion
    depends_on("py-poetry-core@1:", type="build")
```

File: var/spack/repos/builtin/packages/py-reportseff/package.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    depends_on("py-click@6.7:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-reportseff/package.py:19]: Missing dep:
```python
depends_on("py-importlib-metadata@4.8.2:4", when="^python@:3.7", type=("build", "run"))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**